llvm/lib/Target/AIE/Utils/AIELoopUtils.h

```diff
@@ -48,6 +48,10 @@ getSingleBlockLoopMBBs(const MachineFunction &MF);
 /// Check if this block is a single block loop.
 bool isSingleMBBLoop(const MachineBasicBlock *MBB);
 
+/// Considering that MBB has a single predecessor that is a loop
+/// and also layout predecessor, return it.
```

Comment:
Actually, it can return a layout predecessor that is not a loop, a unique predecessor that is not a loop or a null pointer (which is also not a loop)

Reply:
This was just a refactoring. Any free use is dangerous, so I put this comment. Maybe we should name it as `getLayoutPredecessor` and then we don't care if it is a loop or not. In this case, it is not a loop utils function anymore.... Any suggestion?